conda_forge_tick/make_graph.py

```diff
@@ -326,136 +317,15 @@ def make_graph(names: List[str], gx: Optional[nx.DiGraph] = None) -> nx.DiGraph:
-                        if m:
-                            pr_id = int(m.group(1))
-                        else:
-                            pr_id = object()
-                    else:
-                        pr_id = object()
-                    pr_info_ordered[pr_id] = (pr_json, node_id, i)
-
-        for pr_id, v in pr_info_ordered.items():
-            if v:
-                (pr_json, node_id, i) = v
-                future = pool.submit(update_function, ghctx, pr_json, gh, dry_run)
-                futures[future] = (node_id, i)
-
-        for f in as_completed(futures):
-            name, i = futures[f]
-            try:
-                res = f.result()
-                if res:
-                    succeeded_refresh += 1
-                    with gx.nodes[name]["payload"] as node:
-                        node["PRed"][i]["PR"].update(**res)
-                        # XXX: This is a bit of a hack
-                        if update_function == close_out_labels:
-                            node["PRed"][i]["data"]["bot_rerun"] = time.time()
-                    logger.info(f"Updated json for {name}: {res['id']}")
-            except github3.GitHubError as e:
-                logger.error(f"GITHUB ERROR ON FEEDSTOCK: {name}")
-                failed_refresh += 1
-                if is_github_api_limit_reached(e, gh):
-                    break
-            except github3.exceptions.ConnectionError:
-                logger.error(f"GITHUB ERROR ON FEEDSTOCK: {name}")
-                failed_refresh += 1
-            except Exception:
-                logger.critical(
-                    "ERROR ON FEEDSTOCK: {}: {}".format(
-                        name, gx.nodes[name]["payload"]["PRed"][i],
-                    ),
-                )
-                raise
-    return succeeded_refresh, failed_refresh
-
-
-def update_graph_pr_status(gx: nx.DiGraph, dry_run: bool = False) -> nx.DiGraph:
-    succeeded_refresh, failed_refresh = _update_pr(refresh_pr, dry_run, gx)
-
-    logger.info(f"JSON Refresh failed for {failed_refresh} PRs")
-    logger.info(f"JSON Refresh succeed for {succeeded_refresh} PRs")
-    return gx
-
-
-def close_labels(gx: nx.DiGraph, dry_run: bool = False) -> nx.DiGraph:
-    succeeded_refresh, failed_refresh = _update_pr(close_out_labels, dry_run, gx)
-
-    logger.info(f"bot re-run failed for {failed_refresh} PRs")
-    logger.info(f"bot re-run succeed for {succeeded_refresh} PRs")
-    return gx
+def update_nodes_with_bot_rerun(gx):
+    """Go through all the open PRs and check if they are rerun"""
+    for name, node in gx.nodes.items():
+        payload = node['payload']
+        for migration in payload.get('PRed', []):
+            pr_json = migration.get('PR', {})
+            # if there is a valid PR and it isn't currently listed as rerun but the PR needs a rerun
+            if pr_json and not migration['bot-rerun'] and "bot-rerun" in [l["name"] for l in pr_json.get("labels", [])]:
```

Comment:
What does the second part of the if statement do here?

Reply:
Check if the PR is labeled with bot rerun

Reply:
Is it checking in both the top level and in the data key? Is the first for backwards compatibility? We only write to the data key now I thought

Reply:
Ah sorry do you mean the `not migration['bot-rerun']` part? That part is checking that we don't already have a rerun on the PR.

Reply:
Yeah so we use data:bot-rerun not the top level key.